internal/pkg/service/stream/aggregation/repository/repository.go

```diff
@@ -0,0 +1,198 @@
+package repository
+
+import (
+	"context"
+	"slices"
+	"strings"
+
+	etcd "go.etcd.io/etcd/client/v3"
+	"golang.org/x/exp/maps"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/log"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/iterator"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/op"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/ptr"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/utctime"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/definition"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/definition/key"
+	definitionRepo "github.com/keboola/keboola-as-code/internal/pkg/service/stream/definition/repository"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/model"
+	storageRepo "github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/repository"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/statistics"
+	statsRepo "github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/statistics/repository"
+)
+
+type dependencies interface {
+	Logger() log.Logger
+	EtcdClient() *etcd.Client
+	DefinitionRepository() *definitionRepo.Repository
+	StatisticsRepository() *statsRepo.Repository
+	StorageRepository() *storageRepo.Repository
+}
+
+type Repository struct {
+	logger     log.Logger
+	client     etcd.KV
+	definition *definitionRepo.Repository
+	statistics *statsRepo.Repository
+	storage    *storageRepo.Repository
+}
+
+type SourceWithSinks struct {
+	SourceKey key.SourceKey
+	Sinks     []*SinkWithStatistics
+}
+
+type SinkWithStatistics struct {
+	*definition.Sink
+	Statistics *SinkStatistics
+}
+
+type SinkStatistics struct {
+	Total *statistics.Aggregated
+	Files []*FileWithStatistics
+}
+
+type FileWithStatistics struct {
+	*model.File
+	Statistics *statistics.Aggregated
+}
+
+func New(d dependencies) *Repository {
+	return &Repository{
+		logger:     d.Logger().WithComponent("aggregation.repository"),
+		client:     d.EtcdClient(),
+		definition: d.DefinitionRepository(),
+		statistics: d.StatisticsRepository(),
+		storage:    d.StorageRepository(),
+	}
+}
+
+func (r *Repository) GetSourcesWithSinksAndStatistics(ctx context.Context, sourceKeys []key.SourceKey) ([]*SourceWithSinks, error) {
```

Comment:
In go we have `getter` using leading character uppercase. Therefore I would expect `SourcesWithSinksAndStatistics` to be a getter. No `Get` needed. 
Also return value is called the same so it looks convinient. :slightly_smiling_face: